### PR TITLE
chore: add a stale bot to manage issues

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

This PR:

- Adds a stale bot to manage issues

## Note

Please make sure to install this [GitHub App](https://github.com/marketplace/stale) to the organization before merging this PR.

### CODE REVIEW

File: .github/stale.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Issues with these labels will never be considered stale
+exemptLabels:
+  - bug
+  - needs-triaging
+  - should-be-fixed
+# Label to use when marking an issue as stale
+staleLabel: wontfix
+# Comment to post when marking an issue as stale. Set to `false` to disable
+markComment: >
+  This issue has been automatically marked as stale because it has not had
```
Comment (silv-io) [inline .github/stale.yml:14]: Maybe add a friendly question to confirm whether this still happens and more information about reproducing the issue

File: .github/stale.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Number of days of inactivity before an issue becomes stale
+daysUntilStale: 60
+# Number of days of inactivity before a stale issue is closed
+daysUntilClose: 7
```
Comment (silv-io) [inline .github/stale.yml:4]: I'd give a bit more time to react to the stale-bot, maybe 2 weeks instead of one. But that's just a gut feeling

**Review by silv-io — COMMENTED** [review]
LGTM :) maybe a better explanation in the message what we're trying to do with stale issues so that people don't get scared away thinking we ignore the issues.

File: .github/workflows/stale-bot.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        stale-issue-message: Hello 👋! It looks like this issue hasn’t been active in longer than two months. 
+          We encourage you to check if this is still an issue in the latest release. 
+          In the absence of more information, we will be closing this issue soon. 
+          If you find that this is still a problem, please feel free to provide a comment or upvote 
+          with a reaction on the initial post to prevent automatic closure. If the issue is already closed, 
+          please feel free to open a new one.
+        # These labels are required
+        stale-issue-label: closing-soon
+        exempt-issue-labels: bug, needs-triaging, should-be-fixed
+        response-requested-label: response-requested
```
Comment (whummer) [inline .github/workflows/stale-bot.yml:30]: Maybe also add `awaiting-response` here? (which is a label that has already been attached to a couple of issues). Or do we deliberately want to create a new label here, to start with a fresh status for the workflow of all tickets?

File: .github/workflows/stale-bot.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          please feel free to open a new one.
+        stale-issue-message: Hello 👋! It looks like this issue hasn’t been active in longer than two months. 
+          We encourage you to check if this is still an issue in the latest release. 
+          In the absence of more information, we will be closing this issue soon. 
+          If you find that this is still a problem, please feel free to provide a comment or upvote 
+          with a reaction on the initial post to prevent automatic closure. If the issue is already closed, 
+          please feel free to open a new one.
+        # These labels are required
+        stale-issue-label: closing-soon
+        exempt-issue-labels: bug, needs-triaging, should-be-fixed
```
Comment (whummer) [inline .github/workflows/stale-bot.yml:29]: Wondering if `should-be-fixed` should be in the exempt list - seems like these are types of issues we deliberately want to close after some time if there's no response from the user (similar to `awaiting-response`)?

Comment (whummer) [inline .github/workflows/stale-bot.yml:29]: Wondering if `should-be-fixed` should be in the exempt list - seems like these are types of issues we deliberately want to close after some time if there's no response from the user (similar to `awaiting-response`)?

File: .github/workflows/stale-bot.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        # Issue timing
+        days-before-stale: 60
+        days-before-close: 14
+        days-before-ancient: 150
+
+        # If you don't want to mark a issue as being ancient based on a
+        # threshold of "upvotes", you can set this here. An "upvote" is
+        # the total number of +1, heart, hooray, and rocket reactions
+        # on an issue.
+        minimum-upvotes-to-exempt: 2
```
Comment (whummer) [inline .github/workflows/stale-bot.yml:45]: Will be interesting to see how this looks like in practice. I think the threshold may be a bit low - but we can bootstrap it with `2` for now, and then potentially increase to `5` or so.. 👍

Comment (whummer) [inline .github/workflows/stale-bot.yml:45]: Will be interesting to see how this looks like in practice. I think the threshold may be a bit low - but we can bootstrap it with `2` for now, and then potentially increase to `5` or so.. 👍

**Review by whummer — APPROVED** [review]
Awesome - long overdue, thanks for taking the lead on this @HarshCasper . Looking forward to seeing this in action! 🚀 

Just a few questions for clarification - nothing too critical. We can also see how it behaves , and then fine-tune the settings as we got.. 👍

File: .github/workflows/stale-bot.yml (hunk 4 above)
Comment (thrau) [reply .github/workflows/stale-bot.yml:29]: +1

**Review by thrau — COMMENTED** [review]
can't wait to see this in action. please check out the comments regarding the labels :-) i think then we're good to go

**Comment by thrau** [comment]
i deleted duplicate comments

File: .github/workflows/stale-bot.yml (hunk 3 above)
Comment (whummer) [reply .github/workflows/stale-bot.yml:30]: Ah, so this was my misunderstanding - thought the purpose was to map this also existing labels (but as the singular name `response-requested-label` implies this should be a single one). Makes sense to rename `response-requested` with `awaiting-response` in the config here (or alternatively leave the PR as-is and switch `awaiting-response`->`response-requested` in the existing issue backlog). 👍

**Comment by HarshCasper** [comment]
The CircleCI tests are failing due to an unrelated change. Hence merging this 🤞

**Pull request merged**